validator/beacon/service.go

```diff
@@ -280,7 +286,10 @@ func (s *Service) PublicKey() []byte {
 // CurrentBeaconSlot based on the genesis timestamp of the protocol.
 func (s *Service) CurrentBeaconSlot() uint64 {
 	secondsSinceGenesis := time.Since(s.genesisTimestamp).Seconds()
-	return uint64(math.Floor(secondsSinceGenesis / params.DefaultConfig().SlotDuration))
+	if math.Floor(secondsSinceGenesis/float64(params.DefaultConfig().SlotDuration))-1 < 0 {
+		return 0
+	}
+	return uint64(math.Floor(secondsSinceGenesis/params.DefaultConfig().SlotDuration)) - 1
```

Comment:
I would avoid using `math.Floor` (I would avoid using floats as a general rule in this entire codebase). 

`return uint64(secondsSinceGenesis)/params.DefaultConfig().SlotDuration` is much safer.

Reply:
I'm also not following why this is being subtracted by 1, although this looks intentional so I'm sure there's a good reason for it.

Reply:
Fixed